### ISSUE TITLE
Support boost_histogram/hist's transformed axis

### ISSUE THREAD

**Issue by goi42** [scikit-hep/uproot5#651]
If I have a [`hist`](https://github.com/scikit-hep/hist) histogram with a [transformed](https://hist.readthedocs.io/en/latest/user-guide/axes.html?highlight=transform#regular-axis-transforms) `Regular` axis, `uproot` does not recognize the transformation and saves the wrong bin edges, as though they were spaced evenly. (There is no such problem for a `Variable` axis.)
___
Reproducer (v4.3.3):
```python
import hist
import matplotlib.pyplot as plt
import numpy as np
import uproot

# without uproot
h_reg = hist.Hist(hist.axis.Regular(19, 0.1, 1e4, transform=hist.axis.transform.log))
h_var = hist.Hist(hist.axis.Variable(np.logspace(-1, 4, 20)))
h_reg.fill([0.15, 1.5, 15, 150, 1500, 15000])
h_var.fill([0.15, 1.5, 15, 150, 1500, 15000])
fig, ax = plt.subplots()
h_reg.plot(ax=ax)
h_var.plot(ax=ax)
plt.xscale("log")
plt.savefig("without_uproot.pdf")

# with uproot
with uproot.recreate("test.root") as f:
    f["h_reg"] = h_reg
    f["h_var"] = h_var
with uproot.open("test.root") as f:
    h_reg = f["h_reg"].to_hist()
    h_var = f["h_var"].to_hist()
fig, ax = plt.subplots()
h_reg.plot(ax=ax)
h_var.plot(ax=ax)
plt.xscale("log")
plt.savefig("with_uproot.pdf")

print(uproot.__version__)
```
[without_uproot.pdf](https://github.com/scikit-hep/uproot5/files/9121434/without_uproot.pdf)
[with_uproot.pdf](https://github.com/scikit-hep/uproot5/files/9121435/with_uproot.pdf)

**Comment by goi42** [comment]
Maybe this helps?

```python
import hist
h_t = hist.Hist(hist.axis.Regular(3, 1, 1e4, transform=hist.axis.transform.log))
a = h_t.axes[0]
print(a.transform)
print(type(a.transform))
print(bool(a.transform))

h_nt = hist.Hist(hist.axis.Regular(3, 1, 1e4))
a = h_nt.axes[0]
print(a.transform)
print(type(a.transform))
print(bool(a.transform))
```
output:
```text
log
<class 'boost_histogram.axis.transform.Function'>
True
None
<class 'NoneType'>
False
```

**Comment by jpivarski** [comment]
This is a feature request, similar to #672, but the scope is smaller: only the boost-histogram/hist → ROOT TH* direction needs to be considered. A transformation of an axis should be evaluated at the time of conversion and variable bin edges should be hard-coded into the ROOT file (because TH* has no concept of transformed axes).

**Comment by goi42** [comment]
"feature" is probably more descriptive, but it's worth emphasizing that a naive user (like myself) is likely to get the wrong values for the bin edges without ever seeing a warning or encountering an error.

**Comment by jpivarski** [comment]
Good point. I've implemented it in #675 (targeting 5.0.0rc2) and #676 (targeting 4.3.5).
